### DISCUSSION TITLE
PrisonBuilding attibutes

### DISCUSSION THREAD

**Discussion by TiagoFar78** [TiagoFar78/PrisonEscape#122] [Polls]
As PrisonBuilding is right now, we store as attributes every location. Every time the game needs a certain location it can be obtained by just calling the getter method of the attribute that stores that location. 
Example: 
When a new instance of PrisonBuilding is created it will store automatically the location for prisioner 1 spawnpoint adding the reference block. Every time player 1 needs to be sent to his cell this location will be gotten by calling some method that will return the desired location.

However, I have been thinking and I considered the possibility of just have the reference block as an attribute, and every time some location is needed it will get the location in the config and add the reference block. (the config is loaded in memory, not in the disk).  Example: 
When a new instance of PrisonBuilding is created it will not store the location for prisioner 1 spawnpoint, instead it will store only the reference block location. Every time player 1 needs to be sent to his cell this location will be gotten by calling some getter method that will get the location from the config (memory) and then convert it to this instance of PrisonBuilding by adding the reference block.

I thought about this because there are a lot of locations and storing them might be consuming some space in memory and is making code difficult to understand. However, the second option loses in time performance.

**Comment by iquelli** [comment]
Maybe when we finish implementation of the first version, we can check how much memory it is consuming. If it's too much we change for version 2
